main.js

```diff
@@ -70,7 +70,7 @@ function main() {
         let buildNumber = fs.readFileSync(path);
         console.log(`Build number already generated in earlier jobs, using build number ${buildNumber}...`);
         //Setting the output and a environment variable to new build number...
-        fs.writeFileSync(process.env.GITHUB_ENV, `BUILD_NUMBER=${buildNumber}`);
+        fs.writeFileSync(process.env.GITHUB_OUTPUT, `BUILD_NUMBER=${buildNumber}`);
         console.log(`::set-output name=build_number::${buildNumber}`);
```

Comment:
I am not really experienced writing own action plugins, but is the log message not the thing that need to be changed? or does the fs.writeFileSync call set the actual variable? as far as i know the GITHUB_ENV and the output variable are two different things.

event if the log output is only for debugging, it should maybe replaced with the new call
Not sure, but should it now be something like this:
```suggestion
        fs.writeFileSync(process.env.GITHUB_ENV, `BUILD_NUMBER=${buildNumber}`);
        fs.writeFileSync(process.env.GITHUB_OUTPUT, `build_number=${buildNumber}`);
```

The same would be true for the change in line 133